sql/catalyst/src/main/scala/org/apache/spark/sql/catalyst/expressions/jsonExpressions.scala

```diff
@@ -554,18 +554,28 @@ case class JsonToStructs(
   @transient
   lazy val converter = nullableSchema match {
     case _: StructType =>
-      (rows: Seq[InternalRow]) => if (rows.length == 1) rows.head else null
+      (rows: Iterator[InternalRow]) => if (rows.hasNext) rows.next() else null
     case _: ArrayType =>
-      (rows: Seq[InternalRow]) => rows.head.getArray(0)
+      (rows: Iterator[InternalRow]) => if (rows.hasNext) rows.next().getArray(0) else null
     case _: MapType =>
-      (rows: Seq[InternalRow]) => rows.head.getMap(0)
+      (rows: Iterator[InternalRow]) => if (rows.hasNext) rows.next().getMap(0) else null
   }
 
-  @transient
-  lazy val parser =
-    new JacksonParser(
-      nullableSchema,
-      new JSONOptions(options + ("mode" -> FailFastMode.name), timeZoneId.get))
+  @transient lazy val parser = {
+    val parsedOptions = new JSONOptions(options, timeZoneId.get)
+    val mode = parsedOptions.parseMode
+    require(mode == PermissiveMode || mode == FailFastMode,
```

Comment:
I think we should move this verification into the constructor.

Reply:
Also, can we use `AnalysisException` instead of `require`?

Reply:
I didn't put `require` to the constructor body directly because of `timeZoneId`. If I move the checking up, I need to move `val parsedOptions = new JSONOptions(options, timeZoneId.get)` too (lazy or not lazy). Checking will force getting of `timeZoneId.get` which will raise an exception. I will check this today or tomorrow.

Reply:
ok, thansk!